lib/storage/azure.js

```diff
@@ -0,0 +1,263 @@
+        options.replicateClusters = []
+      }
+      options.replicateClusters.push({
+        account : options.account,
+        key : options.key,
+        container : options.container,
+      })
+      async.each(options.replicateClusters, function(cluster, cb) {
+        createContainer(cluster, options, function(err, svc) { 
+          if (err) {
+            return cb(err)
+          }
+          blobSvcs.push({
+            svc : svc,
+            container : cluster.container || options.container,
+          })
+          return cb()
+        })
+      }, callback);
+    },
+
+    copyIn: function(localPath, path, options, callback) {
+      gzipme(localPath, false, 'best', function() {
+        path = path[0] === '/' ? path.slice(1) : path;
+        async.each(blobSvcs, function(blobSvc, cb) {
+          createContainerBlob(blobSvc, path, localPath, function(err) {
+            if (err) {
+              __log('! Cannot create blob in container : ' + err)
+            }
+            cb(err)
+          })
+        }, function(err) {
+          if (err) {
+            __log('Cannot create blobs in container: ' + err)
+            return callback(err)
+          }
+          removeLocalBlob(localPath + '.gz', callback)
+        })
+      })
+    },
+
+    copyOut: function(path, localPath, options, callback) {
+      var blob = blobSvcs[0]
+      path = path[0] === '/' ? path.slice(1) : path;
+
+      blob.svc.getBlobToLocalFile(blob.container, path, localPath, function(error, result, response) {
+        if (error) {
+          __log('error azure download file', error);
+          return callback(error);
+        }
+        
+        response.localPath = localPath;
+        
+        var returnVal = {
+          result: result,
+          response: response
+        };
+        
+        return callback(null, returnVal);
+      });
+    },
+
+    remove: function(path, callback) {
+      path = path[0] === '/' ? path.slice(1) : path;
+      
+      async.each(blobSvcs, function(blobSvc, cb) {
+        removeContainerBlob(blobSvc, path, cb)
+      }, callback)
+    },
+
+    disable: function(path, callback) {
+      const blob = blobSvcs[0];
+      const dPath = utils.getDisabledPath(path, self.options.disabledFileKey);
+      self.copyBlob(path, dPath, e => {
+        // if copy fails, abort
+        if (e) {
+          return callback(e);
+        } else { // otherwise, remove original file (azure does not currently support rename operations, so we dance)
+          self.remove(path, callback);
+        }
+      });
+    },
+    
+    enable: function(path, callback) {
+      const blob = blobSvcs[0];
+      const dPath = utils.getDisabledPath(path, self.options.disabledFileKey);
+      self.copyBlob(dPath, path, e => {
+        if (e) {
+          return callback(e);
+        } else {
+          self.remove(dPath, callback); 
+        }
+      });
+    },
+
+    copyBlob: function(_src, dst, callback) {
```

Comment:
Write this as a "private method" (a nested function) so it's not exported.

Reply:
✓